pkg/notary/authenticating_round_tripper.go

```diff
@@ -0,0 +1,119 @@
+package notary
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	"github.com/pkg/errors"
+)
+
+type AuthenticatingRoundTripper struct {
+	Token               string
+	Keychain            authn.Keychain
+	WrappedRoundTripper http.RoundTripper
+}
+
+func (a *AuthenticatingRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
```

Comment:
I'm a little confused by this. It is using the docker credentials to authenticate to the notary server?

Reply:
That is correct. The notary server uses the same auth that docker uses.